File: daq/slc/extra/backup.javalib/src/b2dqm/core/LongArray.java
package b2dqm.core;

import b2daq.core.Reader;
import b2daq.core.Writer;

public class LongArray extends NumberArray {

	private long[] _value_v;
	
	public LongArray() {
		_value_v = new long [0];
		for (int n = 0; n < length(); n++ ) _value_v[n] = 0;
	}

	public LongArray(int len) {
		_value_v = new long [len];
		for (int n = 0; n < length(); n++ ) _value_v[n] = 0;
	}

	public LongArray(NumberArray array) { copy(array); }

	public void copy(NumberArray array) {
		if( _value_v.length != array.length() ) _value_v = new long [array.length()];
		for (int n = 0; n < length(); n++ ) _value_v[n] = (int)array.get(n);
	}

	public void copy(int [] array) {
		if( _value_v.length != array.length ) _value_v = new long [array.length];
		for (int n = 0; n < length(); n++ ) _value_v[n] = array[n];
	}

	public double get(int index) {
		if ( index >= 0 && index < length() ) return _value_v[index];
		else return -1;
	}

	public void set(int index, double data) {
		if ( index >= 0 && index < length() ) _value_v[index] = (int)data;
	}

	public void set(int index, long data) {
		if ( index >= 0 && index < length() ) _value_v[index] = data;
	}

	public int length() { return _value_v.length; }

	public void resize(int len) {
		_value_v = new long [len];
		for (int n = 0; n < length(); n++ ) _value_v[n] = 0;
	}

	public void writeSingle(double value, Writer writer) throws Exception {
		writer.writeLong((int)value);
	}

	public double readSingle(Reader reader) throws Exception {
		return reader.readLong();
	}

}

File: daq/slc/dqm/src/LongArray.cc
#include "daq/slc/dqm/LongArray.h"

#include <sstream>

using namespace Belle2;

LongArray::LongArray() throw() : NumberArray(0), m_value_v(NULL) {}

LongArray::LongArray(size_t size) throw() : NumberArray(), m_value_v(NULL)
{
  resize(size);
}

LongArray::LongArray(const NumberArray& v) throw() : NumberArray(), m_value_v(NULL)
{
  resize(v.size());
  for (size_t n = 0; n < size(); n++) set(n, m_value_v[n]);
}

LongArray::~LongArray() throw()
{
  if (m_value_v != NULL) delete [] m_value_v;
}

double LongArray::get(size_t n) const throw()
{
  if (n < size()) return m_value_v[n];
  else return -1;
}
void LongArray::set(size_t n, double v) throw()
{
  if (n < size()) m_value_v[n] = (long long)v;
}
void LongArray::resize(int s) throw()
{
  if (m_value_v != NULL) delete [] m_value_v;
  m_value_v = new long long [s];
  m_size = s;
  for (size_t n = 0; n < size(); n++) set(n, 0);
}

std::string LongArray::toScript() const throw()
{
  std::stringstream ss("");
  for (size_t n = 0; n < size(); n++) {
    ss << get(n);
    if (n < size() - 1) ss << ",";
  }
  return ss.str();
}

long long& LongArray::operator[](size_t n) throw()
{
  if (n < size()) return m_value_v[n];
  return m_value_v[0];
}

long long LongArray::operator[](size_t n) const throw()
{
  if (n < size()) return m_value_v[n];
  return 0;
}

void LongArray::readObject(Reader& reader)
throw(IOException)
{
  for (size_t n = 0; n < size(); n++) {
    m_value_v[n] = reader.readLong();
  }
}

void LongArray::writeObject(Writer& writer)
const throw(IOException)
{
  for (size_t n = 0; n < size(); n++) {
    writer.writeLong(m_value_v[n]);
  }
}

double LongArray::readElement(Reader& reader)
throw(IOException)
{
  return reader.readLong();
}

void LongArray::writeElement(double value, Writer& writer)
const throw(IOException)
{
  writer.writeLong(value);
}
